### PR TITLE
improve keyboard shortcuts

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/392891325557410/1183224745168980
Tech Design URL:
CC:

**Description**:

Add new keyboard shortcuts:

* ⌘G / ⌘⇧G - Find next / previous
* ⌘R - Reload
* ⌘ Click / ⌘⇧ Click - Open in new background/ foreground tab (requires iOS 13.4)
* left/right arrows for navigating tab switcher (not showing in shortcuts help)
* ⌘ ⌥ Tab - Open tab switcher
* ⌘ P - Print
* ⌘ D - Add Bookmark
* ⌘ shift D - Add Favorite 

Fixes a bug when using keyboard shortcuts to navigate tabs.

**Steps to test this PR**:
1. Press and then press and hold down ⌘ to show the shortcuts screen.  Make sure it looks complete.
1. Using an iPad + keyboard  (could be simulator with "I/O > Input > Send Keyboard Input to Device" enabled, try out the above commands.
1. Ensure previous commands still work
1. Open tab switcher and confirm help screen looks correct and commands still work as expected
1. Open a few tabs, leave a couple on the home screen, and browse on the others, ensure that next / previous tab work as expected.

**Device Testing**:

* [ ] iPad

**OS Testing**:

* [ ] iOS 13

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by brindy** [comment]
This has been approved in Product Review.

File: DuckDuckGo/MainViewController+KeyCommands.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ extension MainViewController {
         }
         
         var browsingCommands = [UIKeyCommand]()
-        if currentTab != nil {
+        if homeController == nil {
```
Comment (bwaresiak) [inline DuckDuckGo/MainViewController+KeyCommands.swift:36]: This is very good change - I think I recall seeing some crashes related to the `currentTab` call having a side effects of creating tab instance.

File: DuckDuckGo/MainViewController+KeyCommands.swift [hunk 2]
```diff
@@ -46,11 +46,33 @@ extension MainViewController {
                 UIKeyCommand(input: UIKeyCommand.inputLeftArrow, modifierFlags: [.command], action: #selector(keyboardBrowserBack),
                              discoverabilityTitle: UserText.keyCommandBrowserBack),
                 UIKeyCommand(input: "w", modifierFlags: .command, action: #selector(keyboardCloseTab),
```
Comment (bwaresiak) [inline DuckDuckGo/MainViewController+KeyCommands.swift]: I'd like to be able to close the home  tab with `cmd+w` - what do you think @brindy?

File: DuckDuckGo/TabSwitcherViewController+KeyCommands.swift [hunk 3]
```diff
@@ -26,19 +26,30 @@ extension TabSwitcherViewController {
...
             UIKeyCommand(input: UIKeyCommand.inputUpArrow, modifierFlags: [], action: #selector(keyboardMoveSelectionUp),
                          discoverabilityTitle: UserText.keyCommandPreviousTab),
             UIKeyCommand(input: UIKeyCommand.inputDownArrow, modifierFlags: [], action: #selector(keyboardMoveSelectionDown),
                          discoverabilityTitle: UserText.keyCommandNextTab),
+            
+            UIKeyCommand(input: UIKeyCommand.inputLeftArrow, modifierFlags: [], action: #selector(keyboardMoveSelectionUp),
+                         discoverabilityTitle: UserText.keyCommandPreviousTab),
+            UIKeyCommand(input: UIKeyCommand.inputRightArrow, modifierFlags: [], action: #selector(keyboardMoveSelectionDown),
+                         discoverabilityTitle: UserText.keyCommandNextTab),
+
```
Comment (bwaresiak) [inline DuckDuckGo/TabSwitcherViewController+KeyCommands.swift:50]: This is actually pretty confusing on a grid. We prolly should allow for 2-dimensional navigation there.

File: DuckDuckGo/MainViewController+KeyCommands.swift (hunk 2 above)
Comment (brindy) [reply DuckDuckGo/MainViewController+KeyCommands.swift]: Yep will change that

File: DuckDuckGo/TabSwitcherViewController+KeyCommands.swift (hunk 3 above)
Comment (brindy) [reply DuckDuckGo/TabSwitcherViewController+KeyCommands.swift:50]: I thought about that but Safari does the the same doesn’t it?

Comment (bwaresiak) [reply DuckDuckGo/TabSwitcherViewController+KeyCommands.swift:50]: For me keyboard doesn't work at all with tab switcher on safari. But still, even if that were the case, I think we can do a better job at this than Safari does. :)

File: DuckDuckGo/TabViewController.swift [hunk 4]
```diff
@@ -218,6 +228,25 @@ class TabViewController: UIViewController {
...
+                tapLinkDestination = .newTab
+            } else {
+                tapLinkDestination = .backgroundTab
+            }
+        }
+    }
+    
+    override func pressesEnded(_ presses: Set<UIPress>, with event: UIPressesEvent?) {
+        super.pressesEnded(presses, with: event)
+        tapLinkDestination = .currentTab
```
Comment (bwaresiak) [inline DuckDuckGo/TabViewController.swift]: Try this:
- Press and hold cmd.
- Press and release shift.
- While still holing cmd, tap on a link.

Expected: link opens in new tab.
Actual: link is just opened.

File: DuckDuckGo/TabSwitcherViewController+KeyCommands.swift (hunk 3 above)
Comment (brindy) [reply DuckDuckGo/TabSwitcherViewController+KeyCommands.swift:50]: I looked at this and it's pretty complex for something that's supposed to be a quick improvement.  Safari with an actual keyboard definitely handles arrows the same way.  I can't get the simulator to handle arrow keys in the Safari tab switch though.  Might come back to this if I can wrap up the main part of this project in a timely fashion.

**Review by bwaresiak — APPROVED** [review]
LGTM!

**Pull request merged**